bin/zmonlyofficeconfig

```diff
@@ -0,0 +1,256 @@
+    
+  fi
+
+  content=$(${zmjq} --arg zimbra_mysql_user "$zimbra_mysql_user" '.services.CoAuthoring.sql.dbUser=$zimbra_mysql_user' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  #password
+  config_mysql_pass=$(su - zimbra -c "zmlocalconfig -x -s zimbra_mysql_password")
+  if [ ! -z "$config_mysql_pass" ]; then
+    IFS='='
+    read -ra arr <<< "$config_mysql_pass"
+    zimbra_mysql_password=$(echo "${arr[1]}" | xargs)
+    
+  fi
+
+  content=$(${zmjq} --arg zimbra_mysql_password "$zimbra_mysql_password" '.services.CoAuthoring.sql.dbPass=$zimbra_mysql_password' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+}
+
+update_certificate_path()
+{
+  content=$(${zmjq} --arg server_crt "$server_crt" '.services.CoAuthoring.ssl.cert=$server_crt' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg server_crt "$server_crt" '.SpellChecker.ssl.cert=$server_crt' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg server_key "$server_key" '.services.CoAuthoring.ssl.key=$server_key' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg server_key "$server_key" '.SpellChecker.ssl.key=$server_key' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+}
+
+update_token_config()
+{
+
+  content=$(${zmjq} '.services.CoAuthoring.token.enable.browser=true' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} '.services.CoAuthoring.token.enable.request.inbox=true' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} '.services.CoAuthoring.token.enable.request.outbox=true' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} '.services.CoAuthoring.requestDefaults.rejectUnauthorized=false' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+}
+
+update_jwt_config()
+{
+  jwtsecret=$1
+  cmd="su - zimbra -c \"zmprov -m -l mcf zimbraDocumentEditingJwtSecret ${jwtsecret}\""
+  eval "$cmd"
+  RC=$?
+  if [ $RC -ne "0" ]; then
+    STATUS=$R
+    echo "Updating zimbraDocumentEditingJwtSecret faced problem."
+  fi
+
+  #update the default.json
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.browser.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.inbox.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.outbox.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+
+  content=$(${zmjq} --arg jwtsecret "$jwtsecret" '.services.CoAuthoring.secret.session.string=$jwtsecret' ${defaultjson_path})
+  echo "${content}" > ${defaultjson_path}
+}
+
+set_jwt_secret()
```

Comment:
You are referencing arguments here..But never passed one in the calls.

Reply:
It has been kept so that if someone wants to use this method to set jwt-secret, they can do it. It can be removed too.

Reply:
done